tests/nn/test_inference.py

```diff
@@ -630,6 +632,34 @@ def test_single_instance_predictor_high_peak_thresh(
     assert len(labels_pr[1]) == 0
 
 
+def test_centroid_predictor(min_labels, min_centroid_model_path):
+    # Load model path
+
+    predictor = CentroidPredictor.from_trained_models(
+        model_path=min_centroid_model_path
+    )
+    predictor.verbosity = "none"
+    labels_pr = predictor.predict(min_labels)
+    assert len(labels_pr) == 1
+    assert len(labels_pr[0].instances) == 2
+
+    assert predictor.is_grayscale == True
+
+    points_gt = np.concatenate(
+        [min_labels[0][0].numpy(), min_labels[0][1].numpy()], axis=0
+    )
+    points_pr = np.concatenate(
+        [labels_pr[0][0].numpy(), labels_pr[0][1].numpy()], axis=0
+    )
+    # points_pr = np.nan_to_num(points_pr, nan=-1000)
+    points_pr = points_pr[~np.isnan(points_pr).any(axis=1)]
+    # points_pr_centroid = [point for point in points_pr if not np.any(np.isnan(point))]
+    inds1, inds2 = sleap.nn.utils.match_points(points_gt, points_pr)
+    points_gt_closest = points_gt[inds1.numpy()][2:]
+    points_pr = points_pr[inds2.numpy()][2:]
+    assert_allclose(points_gt, points_gt_closest, atol=40)
```

Comment:
This test fails for two reasons... 
1. the coordinates returned in the centroid predictor are assigned to just the first node of the instance
2. if an anchor point is not specified, then the midpoint of the bounding box is used as the centroid:
https://github.com/talmolab/sleap/blob/ed77b49164b654bf9e43c29fcc3af73be2f8eb3b/sleap/nn/data/instance_centroids.py#L182-L189